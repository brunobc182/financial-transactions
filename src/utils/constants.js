const TRANSACTION_TYPE = {
  DEBIT: 'DEBIT',
  CREDIT: 'CREDIT',
};

export default TRANSACTION_TYPE;
