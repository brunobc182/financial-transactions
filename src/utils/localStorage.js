const setItem = (name, item) => localStorage.setItem(name, JSON.stringify(item));
const getItem = name => JSON.parse(localStorage.getItem(name));

export { setItem, getItem };
