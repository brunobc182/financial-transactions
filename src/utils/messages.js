const MESSAGES = {
  totalLabel: 'Total: ',
  partialTotalLabel: 'Partial total: ',
};

export default MESSAGES;
