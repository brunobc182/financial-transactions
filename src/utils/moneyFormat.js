const moneyFormat = (number, prefix = '$') => `${prefix} ${number
  .toFixed(2)
  .replace(/(\d)(?=(\d{3})+\.)/g, '$1.')
  .replace(/\.(\d{2})$/, ',$1')}`;

export default moneyFormat;
