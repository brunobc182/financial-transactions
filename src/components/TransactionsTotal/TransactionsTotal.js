import React from 'react';
import styled from 'styled-components';
import { array, node, oneOfType } from 'prop-types';
import { moneyFormat, MESSAGES } from '../../utils';
import colors from '../../theme/colors';

const { totalLabel } = MESSAGES;
const TransactionsTotal = ({ data }) => {
  const total = data.reduce((acc, item) => acc + item.value, 0);

  return (
    <TotalWrapper>
      <Total total={total}>
        {totalLabel}
        {moneyFormat(total)}
      </Total>
    </TotalWrapper>
  );
};

const TotalWrapper = styled.div`
  display: flex;
`;
const Total = styled.p`
  margin: 0;
  font-size: 28px;
  font-weight: bold;
  line-height: 1.5;
  color: ${({ total }) => (total >= 0 ? colors.shamrock : colors.radicalRed)};
`;

TransactionsTotal.propTypes = {
  data: oneOfType([node, array]).isRequired,
};

export default TransactionsTotal;
