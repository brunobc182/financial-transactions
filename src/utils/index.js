import { TRANSACTION_TYPE, LOCAL_STORAGE_NAME } from './constants';
import moneyFormat from './moneyFormat';
import MESSAGES from './messages';
import { getItem, setItem } from './localStorage';

export {
  LOCAL_STORAGE_NAME, TRANSACTION_TYPE, moneyFormat, MESSAGES, getItem, setItem,
};
