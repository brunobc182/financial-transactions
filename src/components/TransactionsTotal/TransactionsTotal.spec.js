import React from 'react';
import renderer from 'react-test-renderer';
import { TRANSACTION_TYPE } from '../../utils';

import TransactionsTotal from './TransactionsTotal';

describe('TransactionsTotal', () => {
  const { CREDIT, DEBIT } = TRANSACTION_TYPE;
  const positiveData = [
    {
      value: 10,
      type: CREDIT,
    },
  ];

  it('should render correctly when the total of velues is positive', () => {
    const tree = renderer.create(<TransactionsTotal data={positiveData} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly when the total of velues is positive', () => {
    const negativeData = [
      {
        value: -10,
        type: DEBIT,
      },
    ];
    const tree = renderer.create(<TransactionsTotal data={negativeData} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
