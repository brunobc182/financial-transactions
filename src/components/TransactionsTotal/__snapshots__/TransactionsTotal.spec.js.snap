// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionsTotal should render correctly when the total of velues is positive 1`] = `
<div
  className="sc-bdVaJa kZAgjJ"
>
  <p
    className="sc-bwzfXH fXCYwK"
  >
    Total: 
    $ 10,00
  </p>
</div>
`;

exports[`TransactionsTotal should render correctly when the total of velues is positive 2`] = `
<div
  className="sc-bdVaJa kZAgjJ"
>
  <p
    className="sc-bwzfXH eXCmRs"
  >
    Total: 
    $ -10,00
  </p>
</div>
`;
